### PR TITLE
Expose a type that represents unset value

### LINKED ISSUE #159: Expose a type that represents Unset (Value with -2 len)
Currently, we only support Option<T> -> None -> -1 (Null) but inserting null is the same as inserting a tombstone as far as I am aware and it's better to do unset in the event the value is not set (undefined).

node has support for automatically handling unset as undefined.

We could expose a type that is identical to Option but serializes to 0 length to support unset, or add a configuration option to make Option<T> use unset instead of null.

Please let me know how you'd like to proceed with this task.

### PR DESCRIPTION
**Pull request by havaker**
Fixes #159 by adding `struct Unset` and `enum MaybeUnset<V: Value>`.

## Pre-review checklist

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [x] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test.
- [x] PR description sums up the changes and reasons why they should be introduced.
- [x] I added appropriate `Fixes:` annotations to PR description.

### CODE REVIEW

File: scylla/src/frame/value.rs [hunk 1]
```diff
@@ -218,6 +227,23 @@ impl<T: Value> Value for Option<T> {
...
+        buf.put_i32(-2);
+        Ok(())
+    }
+}
+
+impl<V: Value> Value for MaybeUnset<V> {
+    fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), ValueTooBig> {
+        match self {
+            MaybeUnset::Set(v) => v.serialize(buf),
+            MaybeUnset::Unset => Unset {}.serialize(buf),
```
Comment (piodul) [inline scylla/src/frame/value.rs:242]: IMO

```rust
Unset.serialize(buf)
```

looks nicer

File: scylla/src/frame/value.rs [hunk 2]
```diff
@@ -520,17 +546,17 @@ impl_batch_values_for_tuple!(T0, T1, T2, T3, T4, T5; 0, 1, 2, 3, 4, 5; 6);
...
+impl_batch_values_for_tuple!(T0, T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11;
                              0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11; 12);
-impl_batch_values_for_tuple!(T0, T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12; 
+impl_batch_values_for_tuple!(T0, T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12;
                              0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12; 13);
-impl_batch_values_for_tuple!(T0, T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13; 
+impl_batch_values_for_tuple!(T0, T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13;
                              0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13; 14);
-impl_batch_values_for_tuple!(T0, T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14; 
+impl_batch_values_for_tuple!(T0, T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14;
```
Comment (piodul) [inline scylla/src/frame/value.rs:559]: Please move whitespace cleanup to a separate commit. It's unrelated to the unset value thing.

File: scylla/src/frame/value_tests.rs [hunk 3]
```diff
@@ -33,6 +33,17 @@ fn option_value() {
     assert_eq!(serialized(null_i32), &(-1_i32).to_be_bytes()[..]);
 }
 
+#[test]
+fn unset_value() {
+    assert_eq!(serialized(Unset {}), &(-2_i32).to_be_bytes()[..]);
```
Comment (piodul) [inline scylla/src/frame/value_tests.rs:38]: `Unset {}` -> `Unset`

**Review by piodul — APPROVED** [review]
Left small nitpicks, looks good overall.

File: scylla/src/frame/value.rs (hunk 1 above)
Comment (havaker) [reply scylla/src/frame/value.rs]: Wow, I didn't know this was possible.
Changed

File: scylla/src/frame/value.rs (hunk 2 above)
Comment (havaker) [reply scylla/src/frame/value.rs:559]: Done

File: scylla/src/frame/value_tests.rs (hunk 3 above)
Comment (havaker) [reply scylla/src/frame/value_tests.rs]: Done

**Review by piodul — APPROVED** [review]
